### PR TITLE
Feat export tailwind theme

### PR DESCRIPTION
**Pull request by ryanwolhuter**
We want to avoid duplicating styles and tailwind configs from the design system across our apps. We can achieve this by using tailwind config [presets](https://tailwindcss.com/docs/presets).

This allows us to still have the flexibility of adding further configs in our other apps, while having the tailwind configs and css variables from the design system available to use.

There were quite a lot of conflicting styles and styles that don't match the current figma in the theme swapping code. Since we aren't using that right now and the dark theme is not fully defined, I have removed the theme swapper code. We can use tailwind's method for dealing with theming as discussed on discord.

In the process of doing this I also found a couple of style bugs which I rectified.

I also changed the usages of the Icon component to prefer using the tailwind colors instead of literal hex string colors.

### CODE REVIEW

File: src/connect/components/sidebar/__snapshots__/sidebar.test.tsx.snap [hunk 1]
```diff
@@ -3,7 +3,7 @@
 exports[`Connect Sidebar Component > should match snapshot 1`] = `
 <DocumentFragment>
   <div
-    class="group/sidebar group expanded flex h-full flex-col overflow-hidden transition-none duration-300 animate-in fade-in bg-neutral-1 shadow-[0px_33px_32px_-16px_rgba(0,0,0,0.10),0px_0px_16px_4px_rgba(0,0,0,0.04)]"
```
Comment (ryanwolhuter) [inline src/connect/components/sidebar/__snapshots__/sidebar.test.tsx.snap:6]: same for the `neutral` colors - updated to match colors from the style guide

File: src/globals.css [hunk 2]
```diff
@@ -167,6 +167,8 @@
     --orange-700: hsl(var(--color-orange-700));
     --orange-800: hsl(var(--color-orange-800));
     --orange-900: hsl(var(--color-orange-900));
+    --shadow-sidebar: 0px 33px 32px -16px rgba(0, 0, 0, 0.1),
```
Comment (ryanwolhuter) [inline src/globals.css:170]: we should start using variables for the shadows once they are included in the style guide. I have created this one for now and will add the others as they become available

File: src/powerhouse/components/tree-view-item/tree-view-item.tsx [hunk 3]
```diff
@@ -80,7 +80,7 @@ export const TreeViewItem: React.FC<TreeViewItemProps> = props => {
                             name="caret"
                             className={twMerge(
                                 open && 'rotate-90',
-                                'ease delay-50 pointer-events-none transition',
```
Comment (ryanwolhuter) [inline src/powerhouse/components/tree-view-item/tree-view-item.tsx:83]: I was quite surprised by this catch from the tailwind eslint plugin. it's strange to me, but it turns out that the shortest delay that tailwind has is 75ms https://tailwindcss.com/docs/transition-delay

File: src/shared-tailwind.config.ts [hunk 4]
```diff
@@ -0,0 +1,141 @@
+import type { Config } from 'tailwindcss';
```
Comment (ryanwolhuter) [inline src/shared-tailwind.config.ts:1]: the theme and plugins are now exported separately so that they can also be used as presets in the other apps

**Review by acaldas — APPROVED** [review]
Very cool!

**Review by gpuente — APPROVED** [review]
LGTM

**Pull request merged**